File: 507-PerfectNumber/507-PerfectNumber.cpp
// Last updated: 3/9/2026, 4:55:45 PM
1class Solution {
2public:
3    bool checkPerfectNumber(int num) {
4        int sum = 0;
5        for (int i = 1; i < num; i++) {
6            if (num % i ==  0) {
7                sum += i;
8            }
9        }
10        return sum == num;
11    }
12};

File: 507-PerfectNumber/507-PerfectNumber.js
// Last updated: 3/9/2026, 4:46:13 PM
1/**
2 * @param {number} num
3 * @return {boolean}
4 */
5var checkPerfectNumber = function(num) {
6    let sum = 0;
7    for(let i = 1; i < num; i++){
8        if(num % i === 0){
9            sum += i;
10        }
11    }
12    return sum === num;
13};
14